File: Easy Problems/Leetcode 3754.cpp
class Solution {
public:
    long long sumAndMultiply(int n) {
        stack<int> st;
        int sum = 0;

        while(n){
            int temp = n % 10;
            if(temp != 0){
                sum += temp;
                st.push(temp);
            }
            n /= 10;
        }
        long long x = 0;
        while(!st.empty()){
            x = x * 10 + st.top();
            st.pop();
        }
        return x * sum;
    }
};

File: Easy Problems/Leetcode 3754.py
class Solution:
    def sumAndMultiply(self, n: int) -> int:
        st = str(n)
        conc = 0
        summ = 0
        for i in st:
            if i != '0':
                summ += ord(i) - ord('0')
                conc = conc * 10 + ord(i) - ord('0')
        return conc * summ